### ISSUE TITLE
Add section on connecting to data sources

### ISSUE THREAD

**Issue by marcushellberg** [vaadin/docs#1975]
One of the most common questions we get from new users is how they can hook up Vaadin to their data. 

I suggest we add a new section on connecting to data sources to the docs, covering the following: 

- Connecting to common databases (move https://vaadin.com/docs/latest/integrations/databases)
- Connecting to APIs: REST, GraphQL,...
- Data access/ORMs: Spring Data, jOOQ, JDBC,...

The section should cover basic usage and provide links for learning more. It should not attempt to replicate documentation for any of the technologies. The purpose of the section is to help developers get their backend hooked up faster and to explicitly show that Vaadin works with different backends.

**Comment by jouni** [comment]
Instead of creating a new section, can we improve the discoverability of the Integrations section more? To me that feels like the correct section title, but it’s likely not easy to discover. I’m thinking that an index page similar to this screenshot, with logos if the things you want to integrate with, would help with that, in addition to making the integrations section more prominent on the main docs landing page.

<img width="1218" alt="Screenshot 2022-12-20 at 10 38 57" src="https://user-images.githubusercontent.com/66382/208621794-1960a01f-682a-42c2-b9cc-667e8689f617.png">

**Comment by marcushellberg** [comment]
Yeah, it might work if we redid the section a bit. Today it's a quite mixed bag of things and I would not intuitively go there to find this kind of info.
